packages/types/src/api-reference/api-reference-configuration.ts

```diff
@@ -0,0 +1,297 @@
+import { z } from 'zod'
+
+import type { TargetId } from '../external/index.ts'
+import { migrateProxyUrl } from './helpers/migrate-proxy-url.ts'
+import { migrateThemeVariables } from './helpers/migrate-theme-variables.ts'
+
+/** Available theme presets for the API reference */
+const ThemeIdEnum = z.enum([
+  'alternate',
+  'default',
+  'moon',
+  'purple',
+  'solarized',
+  'bluePlanet',
+  'deepSpace',
+  'saturn',
+  'kepler',
+  'elysiajs',
+  'fastify',
+  'mars',
+  'none',
+])
+
+/** Valid keys that can be used with CTRL/CMD to open the search modal */
+const SearchHotKeyEnum = z.enum([
+  'a',
+  'b',
+  'c',
+  'd',
+  'e',
+  'f',
+  'g',
+  'h',
+  'i',
+  'j',
+  'k',
+  'l',
+  'm',
+  'n',
+  'o',
+  'p',
+  'q',
+  'r',
+  's',
+  't',
+  'u',
+  'v',
+  'w',
+  'x',
+  'y',
+  'z',
+])
+
+/** Supported integration types */
+const IntegrationEnum = z
+  .enum([
+    'adonisjs',
+    'docusaurus',
+    'dotnet',
+    'elysiajs',
+    'express',
+    'fastapi',
+    'fastify',
+    'go',
+    'hono',
+    'html',
+    'laravel',
+    'litestar',
+    'nestjs',
+    'nextjs',
+    'nitro',
+    'nuxt',
+    'platformatic',
+    'react',
+    'rust',
+    'vue',
+  ])
+  .nullable()
+
+/** Configuration for the OpenAPI/Swagger specification */
+const SpecConfigurationSchema = z.object({
+  /** URL to an OpenAPI/Swagger document */
+  url: z.string().url().optional(),
```

Comment:
```suggestion
  url: z.string().optional(),
```
Probably stay away from the url validation as it will fail